clients/src/main/java/org/apache/kafka/common/utils/ChunkedBytesStream.java

```diff
@@ -290,25 +290,27 @@ public long skip(long toSkip) throws IOException {
 
         // Skip bytes stored in intermediate buffer first
         int avail = count - pos;
-        long bytesSkipped = (avail < remaining) ? avail : remaining;
+        int bytesSkipped = Math.min(avail, (int) remaining);
```

Comment:
I am not sure that this is a good idea to downcast and then perform comparison. Comparison amongst different types is performed by upgrading them to the higher type and then performing comparison. In current approach, we downcasting `remaining` to int is unsafe.

May I suggest an alternative where we upgrade avail to `long` before performing comparison and then converting result to int:` int bytesSkipped = (int) Math.min((long) avail, remaining);`

Reply:
Can you please explain an example where these two variants would have a different result given that `avail` is a positive int?

Reply:
> behaviour is implementation defined

Can you share a reference for this? My understanding is that the Java specifies the behavior in these cases. Perhaps you're thinking of a C or something like that?

> let's say remaining = Integer.MIN_VALUE + 1000

I think you meant to say that if we have a `long` of value (Integer.MAX_VALUE + 1000) may result in sign extension during truncation resulting in a negative int. That's a fair point - my bad. I'll fix this. It does show a gap in the tests for this class, we should ideally have tests that cover these boundary cases.

Reply:
> It does show a gap in the tests for this class, we should ideally have tests that cover these boundary cases.

Fair point. Added a JIRA https://issues.apache.org/jira/browse/KAFKA-15123

Reply:
Thanks!